src/libcore/iter/iterator.rs

```diff
@@ -258,6 +258,32 @@ pub trait Iterator {
         None
     }
 
+    /// Returns an iterator adaptor that goes over the original iterator,
+    /// skipping `step` elements at a time.
+    ///
+    /// Panics if `step == 0`.
+    ///
+    /// # Examples
+    ///
+    /// Basic usage:
+    ///
+    /// ```
+    /// let a = [0, 1, 2, 3, 4, 5];
+    ///
+    /// let mut iter = a.iter().every_nth(2);
+    ///
+    /// assert_eq!(iter.next(), Some(&0));
+    /// assert_eq!(iter.next(), Some(&2));
+    /// assert_eq!(iter.next(), Some(&4));
+    /// assert_eq!(iter.next(), None);
+    /// ```
+    #[inline]
+    #[stable(feature = "rust1", since = "1.0.0")]
+    fn every_nth(self, step: usize) -> EveryNth<Self> where Self: Sized {
+        assert!(step != 0);
```

Comment:
@nagisa more like impossible without cloning (peak returns a reference, `next` would return a value).

From the description ("skipping `step` elements at a time"), I'd expect `every_nth(0)` to return every item and `every_nth(1)` to skip every other item. This would also be consistent with `nth`.